xclim/indices/_agro.py

```diff
@@ -1418,3 +1419,75 @@ def effective_growing_degree_days(
     egdd = aggregate_between_dates(deg_days, start=start, end=end, freq=freq)
 
     return to_agg_units(egdd, tas, op="delta_prod")
+
+
+@declare_units(tasmin="[temperature]")
+def hardiness_zones(
+    tasmin: xarray.DataArray, window: int = 30, method: str = "usda", freq: str = "YS"
+):
+    """Hardiness zones.
+
+    Hardiness zones are a categorization of the annual extreme temperature minima, averaged over a certain period.
+    The USDA defines 14 zones, each divided into two sub-zones, using steps of 5°F, starting at -65°F.
+    The Australian National Botanic Gardens define 7 zones, using steps of 5°C, starting at -15°C.
+
+    Parameters
+    ----------
+    tasmin : xr.DataArray
+        Minimum temperature.
+    window : int
+        The length of the averaging window, in years.
+    method : {'usda', 'anbg'}
+        Whether to return the American (`usda`) or the Australian (`anbg`) classification zones.
+    freq : str
+        Resampling frequency.
+
+    Returns
+    -------
+    xr.DataArray, [dimensionless]
+        {method} hardiness zones.
+        US sub-zones are denoted by using a half step. For example, Zone 4b is given as 4.5.
+        Values are given at the end of the averaging window.
+
+    References
+    ----------
+    :cite:cts:`usda_2012`
+    :cite:cts:`dawson_plant_1991`
+    """
+    tnmin = tasmin.resample(time=freq).min().rolling(time=window).mean()
+
+    if method.lower() == "usda":
+        # Bins in [-65,65] degF with steps of 5 degF
+        b0, b1 = convert_units_to("-65 degF", tnmin), convert_units_to("65 degF", tnmin)
+        bins = np.append(
+            np.insert(np.linspace(b0, b1, int(1 + (2 * 65) / 5)), 0, -np.inf), np.inf
+        )
+
+        def _zones(arr, bs):
+            c = np.digitize(arr, bs).astype(float)
+            c[c == 0] = np.nan
+            c[c == bs.size] = np.nan
+            return (c - 1) / 2
+
+    elif method.lower() == "anbg":
+        # Bins in [-15,20] degC with steps of 5 degC
+        b0, b1 = convert_units_to("-15 degC", tnmin), convert_units_to("20 degC", tnmin)
+        bins = np.append(
+            np.insert(np.linspace(b0, b1, int(1 + (20 + 15) / 5)), 0, -np.inf), np.inf
+        )
+
+        def _zones(arr, bs):
+            c = np.digitize(arr, bs).astype(float)
+            c[c == 0] = np.nan
+            c[c == bs.size] = np.nan
+            return c
+
```

Comment:
Taken USDA example as reference:
* Zone 1a is used for example in Alaska
* I don't see that Zone12a to 13b are used at all. 

For zones colder than 1a := 1rst zone, could we attribute the value 0? 
For zones hotter than 13b:= 26th zone, could we assign the value 27?

With window=30, we probably won't reach such extremes, but window=1 is a possible input, and we have seen temperature extremes (below the -60degF) that would be NaN in this case. Is there a reason it's better to put NaNs?